packages/langium-cli/src/generate.ts

```diff
@@ -4,74 +4,104 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-import * as fs from 'fs-extra';
-import { Package } from './package';
-import { Grammar, createLangiumGrammarServices, LangiumDocumentConfiguration, resolveAllReferences } from 'langium';
+import fs from 'fs-extra';
+import path from 'path';
+import { LangiumConfig, RelativePath } from './package';
+import { Grammar, createLangiumGrammarServices, LangiumDocumentConfiguration } from 'langium';
 import { generateGrammarAccess } from './generator/grammar-access-generator';
 import { generateParser } from './generator/parser-generator';
 import { generateAst } from './generator/ast-generator';
 import { generateModule } from './generator/module-generator';
 import { generateTextMate } from './generator/textmate-generator';
-import path from 'path';
 import { serializeGrammar } from './generator/grammar-serializer';
 
 export type GenerateOptions = {
     file?: string;
 }
 
-export function generate(opts: GenerateOptions): void {
-    const file = opts.file ?? './package.json';
-    const packageContent = fs.readFileSync(file).toString();
-    const pack = <Package>JSON.parse(packageContent);
-
+export function generate(config: LangiumConfig): void {
     const services = createLangiumGrammarServices();
+    const relPath = config[RelativePath];
 
-    const grammarFile = pack.langium.grammar ?? 'src/grammar.langium';
-    const grammarFileContent = fs.readFileSync(grammarFile).toString();
+    const grammarFile = config.grammar ?? 'src/grammar.langium';
```

Comment:
I think we should make the `grammar` property mandatory and drop this default value.